pkg/plugins/platform/local.go

```diff
@@ -830,26 +833,31 @@ func (l *LocalApp) Down(removeData bool) error {
 				}
 			}
 		}
-		dir := filepath.Dir(l.AppConfig.DataDir)
+		// Check that l.AppConfig.DataDir is a directory that is safe to remove.
+		err = validateDataDirRemoval(l.AppConfig)
+		if err != nil {
+			return fmt.Errorf("failed to remove data directories: %v", err)
+		}
 		// mysql data can be set to read-only on linux hosts. PurgeDirectory ensures files
 		// are writable before we attempt to remove them.
-		if !fileutil.FileExists(dir) {
+		if !fileutil.FileExists(l.AppConfig.DataDir) {
 			util.Warning("No application data to remove")
 		} else {
-			err := fileutil.PurgeDirectory(dir)
+			err := fileutil.PurgeDirectory(l.AppConfig.DataDir)
 			if err != nil {
 				return fmt.Errorf("failed to remove data directories: %v", err)
 			}
 			// PurgeDirectory leaves the directory itself in place, so we remove it here.
-			err = os.RemoveAll(dir)
+			err = os.RemoveAll(l.AppConfig.DataDir)
 			if err != nil {
-				return fmt.Errorf("failed to remove data directories: %v", err)
+				return fmt.Errorf("failed to remove data directory %s: %v", l.AppConfig.DataDir, err)
 			}
 			util.Success("Application data removed")
 		}
 	}
 
-	return StopRouter()
+	err = StopRouter()
+	return err
```

Comment:
nit: this seems like an odd change, but doesn't hurt anything.

Reply:
It makes debugging easier. Just a style preference of mine. Makes the intent very clear without looking up what StopRouter() does and returns.